pyproject.toml

```diff
@@ -174,6 +176,41 @@ privacy = [
 project-name = "Temporal Python"
 sidebar-expand-depth = 2
 
+[tool.pyright]
```

Comment:
That's a lot of excluded files. Are we sure we don't want to solve event some of them here?

Reply:
Yes, I'd like to keep this PR focused on the infrastructure, and subsequent PRs focused on actual type checking considerations.